lib/matchers/equals.js

```diff
@@ -0,0 +1,21 @@
+var _ = require('lodash');
+var factory  = require('../factory');
+
+module.exports = factory({
+  initialize: function(expectedValue) {
+    this.expectedValue = expectedValue;
+  },
+  match: function(path, value) {
+    if (typeof this.expectedValue === 'undefined') {
+      return 'should be defined';
+    }
+
+    if (_.isObject(this.expectedValue)) {
+      return 'should be a primitive';
```

Comment:
Yes most matchers throw an exception if the setup if wrong, e.g. `s.array(min: 'foobar')`. It also helps speed things up so it's not checked on every call to `match`.